src/UKHO.ERPFacade.Common/IO/AzureBlobEventWriter.cs

```diff
@@ -1,45 +1,37 @@
-﻿using Azure.Storage.Blobs;
-using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.Options;
-using Newtonsoft.Json.Linq;
-using System.Diagnostics.CodeAnalysis;
+﻿using System.Diagnostics.CodeAnalysis;
 using System.Text;
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Options;
 using UKHO.ERPFacade.Common.Configuration;
-using UKHO.ERPFacade.Common.Logging;
 
 namespace UKHO.ERPFacade.Common.IO
 {
     [ExcludeFromCodeCoverage]
     public class AzureBlobEventWriter : IAzureBlobEventWriter
     {
-        private readonly ILogger<AzureBlobEventWriter> _logger;
         private readonly IOptions<AzureStorageConfiguration> _azureStorageConfig;
 
-        public AzureBlobEventWriter(ILogger<AzureBlobEventWriter> logger,
-                                        IOptions<AzureStorageConfiguration> azureStorageConfig)
+        public AzureBlobEventWriter(IOptions<AzureStorageConfiguration> azureStorageConfig)
         {
-            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
             _azureStorageConfig = azureStorageConfig ?? throw new ArgumentNullException(nameof(azureStorageConfig));
         }
 
-        public async Task UploadEvent(JObject eesEvent, string traceId)
+        public async Task UploadEvent(object requestObject, string requestFormat, string traceId)
         {
-            BlobClient blobClient = GetBlobClient(traceId);
+            BlobClient blobClient = GetBlobClient(traceId, requestFormat);
 
-            Stream stream = new MemoryStream(Encoding.UTF8.GetBytes(eesEvent.ToString() ?? ""));
+            Stream stream = new MemoryStream(Encoding.UTF8.GetBytes(requestObject.ToString() ?? ""));
 
             await blobClient.UploadAsync(stream, overwrite: true);
-
-            _logger.LogInformation(EventIds.UploadedEncContentPublishedEventInAzureBlob.ToEventId(), "ENC content published event is uploaded in blob storage successfully.");
         }
 
         //Private Methods
-        private BlobClient GetBlobClient(string containerName)
+        private BlobClient GetBlobClient(string containerName, string requestFormat)
         {
             BlobContainerClient blobContainerClient = new(_azureStorageConfig.Value.ConnectionString, containerName);
             blobContainerClient.CreateIfNotExists();
 
-            var blobName = containerName + ".json";
+            var blobName = containerName + '.' + requestFormat;
```

Comment:
Can we pass this blob name from calling method so that this method will be generic and won't be depend of request format.

Reply:
Done.